core/src/main/java/com/linecorp/armeria/server/HttpHeaderPathMapping.java

```diff
@@ -0,0 +1,220 @@
+            return result;
+        }
+
+        // Also, the missing '@ProduceTypes' would be handled the same as '@ProduceTypes({"*/*"})'.
+        final List<MediaType> producibleTypes = produceTypes.isEmpty() ? ANY_TYPE : produceTypes;
+        for (MediaType produceType : producibleTypes) {
+            for (int i = 0; i < types.size(); i++) {
+                if (produceType.belongsTo(types.get(i))) {
+                    // To early stop path mapping traversal,
+                    // we set the score as the best score when the index is 0.
+                    result.setScore(i == 0 ? PathMappingResult.HIGHEST_SCORE : -1 * i);
+                    if (!produceTypes.isEmpty()) {
+                        result.setNegotiatedProduceType(produceType);
+                    }
+                    return result;
+                }
+            }
+        }
+
+        mappingCtx.delayThrowable(NOT_ACCEPTABLE_EXCEPTION);
+        return PathMappingResult.empty();
+    }
+
+    @Override
+    public Set<String> paramNames() {
+        return pathStringMapping.paramNames();
+    }
+
+    @Override
+    public String loggerName() {
+        return loggerName;
+    }
+
+    @Override
+    public String metricName() {
+        return metricName;
+    }
+
+    @Override
+    public Optional<String> exactPath() {
+        return pathStringMapping.exactPath();
+    }
+
+    @Override
+    public Optional<String> prefix() {
+        return pathStringMapping.prefix();
+    }
+
+    @Override
+    public Optional<String> triePath() {
+        return pathStringMapping.triePath();
+    }
+
+    @Override
+    public int complexity() {
+        return complexity;
+    }
+
+    public PathMapping pathStringMapping() {
+        return pathStringMapping;
+    }
+
+    public Set<HttpMethod> supportedMethods() {
+        return supportedMethods;
+    }
+
+    public List<MediaType> consumeTypes() {
+        return consumeTypes;
+    }
+
+    public List<MediaType> produceTypes() {
+        return produceTypes;
+    }
+
+    private static String generateName(String delim, String prefix, Set<HttpMethod> supportedMethods,
+                                       List<MediaType> consumeTypes, List<MediaType> produceTypes) {
+        final StringJoiner methodNames = new StringJoiner("_");
+        supportedMethods.forEach(e -> methodNames.add(e.name()));
+        StringJoiner name = new StringJoiner(delim).add(prefix)
+                                                   .add(methodNames.toString());
+
+        // The following three cases should be different to each other.
+        // Each name would be produced as follows if the delimiter is assumed as '/'.
+        //
+        // consumeTypes: text/plain, text/html               -> _text_plain/_text_html
+        // consumeTypes: text/plain, produceTypes: text/html -> _text_plain/text_html_
+        // produceTypes: text/plain, text/html               -> text_plain_/text_html_
```

Comment:
It seems a little obscure to use the position of underscore as a way to tell a media type is for consumption or for production. How about:

    consumes/text_plain/text_html/produces/application_octet-stream
    consumes/_/produces/_